lib/autocomplete.ts

```diff
@@ -0,0 +1,93 @@
+import {dom, DomElementArg, onElem, onKeyElem} from 'grainjs';
+import {MaybeObsArray} from 'grainjs';
+import {BaseMenu, defaultMenuOptions, IMenuOptions, menuItem} from './menu';
+import {IOpenController, IPopupContent, PopupControl, setPopupToFunc} from './popup';
+
+/**
+ * User interface for creating a weasel autocomplete element in DOM.
+ *
+ * Usage:
+ *      const employees = ['Thomas', 'June', 'Bethany', 'Mark', 'Marjorey', 'Zachary'];
+ *      const inputElem = input(...);
+ *      autocomplete(inputElem, employees);
+ */
+export function autocomplete<T>(
+  inputElem: HTMLInputElement,
+  optionArray: MaybeObsArray<string>,
+  options: IMenuOptions = {}
+): HTMLInputElement {
+
+  // Options to pass into the Autocomplete class.
+  const menuOptions: IMenuOptions = {
+    ...defaultMenuOptions,
+    menuCssClass: options.menuCssClass,
+    trigger: [(triggerElem: Element, ctl: PopupControl) => {
+      dom.onElem(triggerElem, 'click', () => ctl.open())
```

Comment:
Do we want also the `Down` key to open (e.g. if we use Tabs to get to the input element)?

Reply:
Should it always open on focus?

I still think yes to open on `Down`, since it closes when something is selected.